Flick/Views/PrivacySwitch.swift

```diff
@@ -0,0 +1,158 @@
+
+        unlockImageView.image = UIImage(named: isPrivate ? "unlockInactive" : "unlockActive")
+        unlockImageView.tintColor = isPrivate ? .lightGray : .offWhite
+    }
+
+    private func setupConstraints() {
+        let horizontalOffset = 14
+
+        lockBackgroundView.snp.makeConstraints { make in
+            make.centerY.equalToSuperview()
+            make.centerX.equalTo(self.snp.leading).offset(horizontalOffset)
+            make.size.equalTo(isPrivate ? activeBackgroundSize : inactiveBackgroundSize)
+        }
+
+        lockImageView.snp.makeConstraints { make in
+            make.centerX.centerY.equalToSuperview()
+            make.size.equalTo(isPrivate ? activeLockSize : inactiveLockSize)
+        }
+
+        unlockBackgroundView.snp.makeConstraints { make in
+            make.centerY.equalToSuperview()
+            make.centerX.equalTo(self.snp.trailing).offset(-horizontalOffset)
+            make.size.equalTo(isPrivate ? inactiveBackgroundSize : activeBackgroundSize)
+        }
+
+        unlockImageView.snp.makeConstraints { make in
+            make.centerX.centerY.equalToSuperview()
+            make.size.equalTo(isPrivate ? inactiveLockSize : activeLockSize)
+        }
+    }
+
+    private func remakeConstraints() {
+        let horizontalOffset = 14
+
+        lockBackgroundView.snp.remakeConstraints { remake in
+            remake.centerY.equalToSuperview()
+            remake.centerX.equalTo(self.snp.leading).offset(horizontalOffset)
+            remake.size.equalTo(isPrivate ? activeBackgroundSize : inactiveBackgroundSize)
+        }
+
+        lockImageView.snp.remakeConstraints { remake in
+            remake.centerX.centerY.equalToSuperview()
+            remake.size.equalTo(isPrivate ? activeLockSize : inactiveLockSize)
+        }
+
+        unlockBackgroundView.snp.remakeConstraints { remake in
+            remake.centerY.equalToSuperview()
+            remake.centerX.equalTo(self.snp.trailing).offset(-horizontalOffset)
+            remake.size.equalTo(isPrivate ? inactiveBackgroundSize : activeBackgroundSize)
+        }
+
+        unlockImageView.snp.remakeConstraints { remake in
+            remake.centerX.centerY.equalToSuperview()
+            remake.size.equalTo(isPrivate ? inactiveLockSize : activeLockSize)
+        }
+    }
+
+    private func update() {
+        UIView.animate(withDuration: 0.5) {
+            self.setupViews()
+            self.remakeConstraints()
+        }
+    }
+
+    func toggle() {
+        self.isPrivate ? self.setPrivate(false) : self.setPrivate(true)
+    }
+
+    func setPrivate(_ isPrivate: Bool) {
+        self.isPrivate = isPrivate
+        delegate?.privacyChanged(isPrivate: isPrivate)
+    }
+
+    override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
+        super.touchesEnded(touches, with: event)
+        self.sendHapticFeedback()
+        self.toggle()
+    }
+
+    // It will give a physical feedback to the user when they tap to toggle.
```

Comment:
oooo! ive been thinking about adding this to the app too! after we finish more screens